### ISSUE TITLE
New Batch Job to Correct DS Expiration Date Issues (SR) (PROD Stabilization)

### ISSUE THREAD

**Issue by TracyLinder** [cityofaustin/atd-data-tech#16894]
### Create a batch job to correct DS folders that have not been issued, but have an expiration date.

**New batch job that:**
- Runs once a week
- ~Updates the folder status to Pending that Reverts the folder status to the last folder status, unless the previous status is Application Incomplete, in which case update it to Pending~
- If the folder status is expired, update the status to the previous folder status. _(All other statuses we will not check the status for updating.)_
- Removes the folder expiration date
- (See [11203](https://github.com/cityofaustin/atd-data-tech/issues/11203) for previously written job)

**The batch job applies to:**
- Folder type is DS
- Folder issue date is null
- Folder expiration date exists

**Comment by TracyLinder** [comment]
1. Folder has not been issued, but an expiration date exists This is the one happening most frequently. 
- Solution is to remove the expiration date.


2. Status is Expired, but expiration date is in the future (2020-026668 DS) (Priya found 33 records on prod)
- Solution is to ??


3. Status is Active, but expiration date is in the past.
- Solution is to ??

@Lpriya-LIT will add her query export to this issue.

**Comment by TracyLinder** [comment]
### Next Steps
- @rajeshlaunch will work with Kyle to determine the best solution for each of the 3 scenarios. Our notes with possible solutions to discuss are above.

**Comment by Lpriya-LIT** [comment]
[16894 - Scenario 2 - 33 records on Prod.xlsx](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBeStHQkE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--528b5765a5e65708a63fe2124626e8713b623a41/16894%20-%20Scenario%202%20-%2033%20records%20on%20Prod.xlsx)

**Comment by rajeshlaunch** [comment]
Notes:

1. Explained all 3 scenarios to Kyle and provided information about each scenario.
2. For Scenario 2 and 3, Raj will share the files to Kyle to review and possibly identify the pattern.
3. Scenario 1 will be a batch process which is to remove expiration dates and Kyle in agreement with this.
4. Next steps: Kyle will review the spreadsheet data and then will provide guidance to take necessary actions on those records as well as future records.


CC: @vschandramouli

**Comment by rajeshlaunch** [comment]
Notes from the meeting with Kyle and Kim:

**2. Status is Expired, but expiration date is in the future (2020-026668 DS) (Priya found 33 records on prod)**

One of the job is updating the expiration date even the folder is expired and the expiration date setup properly. Need to figure out, why?

**3. Status is Active, but expiration date is in the past. (39 records in PROD)**

On May 6th one of the batch updated all these "Active" records to "Expired". We need to figure it out with developer?

**Comment by SubhaShroff** [comment]
@rajeshlaunch  @vschandramouli 
### Status is Active, but expiration date is in the past:
Here is the explanation why all the DS at once got expired: - JobRSN -38 (Expire BMEP Pemits whose expiry data has passed). 
This is the already existing batch on PROD that expires DS  along with BP, MP, EP, PP permits if the permit has past expiry date and not EXPIRED, on daily basis. If you notice in the attached excel, since 25th February, 2024, this batch started giving error and may not be updating the permits' status due to failure. Looks like DSD fixed it or something has changed the batch got revived and started updating the permits since May 7th.
This must be the reason that all 39 records must have reactivated at once on 7th. I checked the expiry date for all those DS , almost all has expiry date after on or after 25th Feb. 
Thank you.
[38 - Batch Instances - Expire BMEP Pemits whose expiry data has passed.csv](https://github.com/cityofaustin/atd-data-tech/files/15264251/38.-.Batch.Instances.-.Expire.BMEP.Pemits.whose.expiry.data.has.passed.csv)

**Comment by TracyLinder** [comment]
### Conversation w/ ROW about scenario 2

- Kim thinks the bug is caused by a system job somewhere that updates expiration dates when an inspection is scheduled on any of the trade permits. 
- Another cause could be when the parent building permit is extended, that job could be updating the DS expiration date.

**Comment by TracyLinder** [comment]
@rajeshlaunch can you please do research for the 2 triggering events described by Kim?

**Comment by TracyLinder** [comment]
**Update**
- We will move forward with the batch job for scenario 1.
- We will move scenarios 2 and 3 to a new card for a quality control report.

**Comment by TracyLinder** [comment]
### Update
- We have refined the scope of this ticket to address only the scenario where an expiration date exists but not an issue date. 
- I've asked Kyle & Kim if there should be a folder status validation, or if the batch should pick up all DS that meet the criteria regardless of folder status.

**Comment by TracyLinder** [comment]
The AC is final. I'm moving this to the backlog. This was not on ROW's prioritized list from August.

**Comment by Nadin-Nader** [comment]
@rajeshlaunch this has been sent to the DevOps CM Team to be deployed to DEV under **SCTASK0454445**

**Comment by Nadin-Nader** [comment]
@rajeshlaunch the new deployment to DEV has been sent to the DevOps CM Team under **SCTASK0455372**

**Comment by Nadin-Nader** [comment]
@rajeshlaunch this has been deployed to DEV please UAT

**Comment by Nadin-Nader** [comment]
This has been sent to the DevOps CM Team to be deployed in DEV under **SCTASK0464920**

**Comment by Nadin-Nader** [comment]
@rajeshlaunch this has been deployed to DEV please UAT

**Comment by rajeshlaunch** [comment]
Deployment plan submitted for TEST

**Comment by Nadin-Nader** [comment]
This has been sent to the DevOps CM Team to be deployed to TEST under **SCTASK0467536**

**Comment by Nadin-Nader** [comment]
@rajeshlaunch Hari mentioned that there is a conflict in your commit. We closed the request. Please fix the conflict and resend to CTM. Priya can assist with the ticket as i will be out on Friday

**Comment by rajeshlaunch** [comment]
@Nadin-Nader I'm in contact with Hari on this.

**Comment by Nadin-Nader** [comment]
This has been sent to the DevOPs CM Team to be deployed to TES under **SCTASK0471851**

**Comment by TracyLinder** [comment]
Hari fixed the DevOps issue and got it in test. Next Raj needs to create a manual request, and ask CTM to create a batch and run it. @Lpriya-LIT @rajeshlaunch please get this done early next week before 11/28 if possible.

**Comment by rajeshlaunch** [comment]
@TracyLinder @Lpriya-LIT Tested in TEST and prod deployment plan is sent to @Nadin-Nader  for Production deployment

**Comment by Nadin-Nader** [comment]
Waiting on Kyle to UAT and Approve the move to PROD

**Comment by Nadin-Nader** [comment]
@KyleErich Have you been able to UAT this on TEST?

**Comment by KyleErich** [comment]
@Nadin-Nader Not yet, you can put it on the Ready for UAT on Test channel. Thank you.

**Comment by Nadin-Nader** [comment]
This has been sent to the DevOps CM Team to be deployed to PROD under **SCTASK0480406**

**Comment by Nadin-Nader** [comment]
@rajeshlaunch @KyleErich this has been deployed to PROD

**Comment by Nadin-Nader** [comment]
@rajeshlaunch  please go ahead and complete the second portion of this

**Comment by KyleErich** [comment]
@Nadin-Nader Does your last post here mean that it's not quite ready for Prod UAT?

**Comment by Nadin-Nader** [comment]
@KyleErich yes sorry for the incomplete update. The first portion is on PROD all that is left to do is to manually create the batch and add the procedures to it.

**Comment by KyleErich** [comment]
@Nadin-Nader No problem, I'll hold off on UAT until it's all in Prod. Thanks.

**Comment by rajeshlaunch** [comment]
@Nadin-Nader User wants to add the status check into this release and it was discussed in Friday call, we need Tracy's direction on this.

**Comment by TracyLinder** [comment]
This is on hold until we get feedback from ROW as to if the AC needs to be updated or not. 
FYI @KyleErich @rajeshlaunch

**Comment by TracyLinder** [comment]
Hi @Lpriya-LIT @Sreelatha-BlazeKannan @Nadin-Nader can one of you please provide an update on this card?

**Comment by Sreelatha-BlazeKannan** [comment]
@TracyLinder  - I started my analysis from yesterday and had a call with Priya to know the status and component details regarding this card.

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader , @TracyLinder - Procedure updates completed and tested in Local; I see that batch job already scheduled in both DEV and TEST; So I would like to promote procedure changes from Local to DEV. So that it can be verified in DEV.

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader - DEV DP Submitted. 

Commit ID: e3686b2ed2fa8ba589eb17d6c3dea89922f49d72

REPORT_EXPIREDSFOLDER

**Comment by Nadin-Nader** [comment]
This has been sent to the DevOps CM Team to be deployed to DEV under **SCTASK0490792**

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader -Deployed to DEV; verified in DEV and testing is in progress.

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader - Tested in DEV all looks good. Prepared test cases and executed the cases. Please find attached Test case document for your reference. 

Here are the 4 cases that I tested - 


- [ ] Expire DS Folder Job (DEV 86028) Executes successfully without errors and sends success email to the email id mentioned in Job when there is data based on Cursor

- [ ] Expire DS Folder Job (DEV 86028) Executes successfully without errors and sends error email to the email id mentioned in Job, even when there is no data as per cursor

- [ ] Expire DS Folder Job Executes successfully Updates expiry date to Null when issue date is Null

- [ ] Expire DS Folder Job Executes successfully. When expiry date is not null and issue date is null. It updates expiry date to Null and also Updates status to 'Pending' Only when status is 'Application incomplete'

[16894 - COA UAT Test Case v1.0.xlsx](https://github.com/user-attachments/files/18950616/16894.-.COA.UAT.Test.Case.v1.0.xlsx)

**Comment by TracyLinder** [comment]
@Sreelatha-BlazeKannan when this gets to test can you please setup and run the same test cases?

**Comment by Nadin-Nader** [comment]
@Sreelatha-BlazeKannan looks like this was deployed today at 10am

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader - I will prepare test cases and test run the job first. And turn the job ON so that we can verify the results on Monday as well.

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader - Prepared Test Cases. Tested in TEST Environment, all looks good. Turned ON the job for tonight's run. 

Here are the 4 cases that I tested and FolderRSN's that I used 13403326,13403327,13402016,13321226,13403328,13403318 -

![Image](https://github.com/user-attachments/assets/0fc708f1-908c-4db6-b3e9-aa3048e71339)

Expire DS Folder Job (DEV 86028) Executes successfully without errors and sends success email to the email id mentioned in Job when there is data based on Cursor


Expire DS Folder Job (DEV 86028) Executes successfully without errors and sends error email to the email id mentioned in Job, even when there is no data as per cursor


Expire DS Folder Job Executes successfully Updates expiry date to Null when issue date is Null


Expire DS Folder Job Executes successfully. When expiry date is not null and issue date is null. It updates expiry date to Null and also Updates status to 'Pending' Only when status is 'Application incomplete'

[16894 - COA UAT TEST- Test Case v1.0.xlsx](https://github.com/user-attachments/files/19061438/16894.-.COA.UAT.TEST-.Test.Case.v1.0.xlsx)

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader - Job Ran last night successfully, here are the results

![Image](https://github.com/user-attachments/assets/ab6cd368-54e9-4d85-b397-62fa3c707f46)

![Image](https://github.com/user-attachments/assets/a40d8a38-1384-4c77-800c-0de87d571440)

**Comment by Nadin-Nader** [comment]
@KyleErich @BondRuss this is ready for UAT on TEST

**Comment by TracyLinder** [comment]
### ROW Review

- We revisited the status rules and the AC was updated. 

_These were working notes:_

Status is `Expired`- Set to previous status

~Status is `Active`- Set to previous status~

Status is `Pending`- Make no status update

Status is `Application Incomplete`- Make no status update

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader - Development is in progress

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader - Development is completed; testing is in progress; however need clarification to test these scenarios as well.

Case 1: When we take previous status from Audit folder, some times previous status is also Expired then what should be the folder status?
Case 2: When we take previous status from Audit folder, some times it is NULL at that time do we need to update as NULL or previous status before NULL ?

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader - Completed testing and created pull request for DEV Promotion. Created task id SCTASK0498529 for the same.

**Commit Details** – 

8a1050ca4deee0c8c14c3b69890113850f6f3dd5: Create a batch job to correct DS folders that have not been issued but have an expiration date. REPORT_EXPIREDSFOLDER.sql 

**Pull Request Link** – https://github.austintexas.gov/AMANDADevOps/Amandav7_TestingPOC/pull/1521

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader - Promoted to DEV and tested in DEV and all looks good.
FolderRSN's I tested - 
11451008
11447311
11074653
11061324
11041346
11029000
11002936
11843368

Before Running the Job

![Image](https://github.com/user-attachments/assets/c2f5b52c-9871-4f40-b415-f0d50abe8313)

Results after running the Job

![Image](https://github.com/user-attachments/assets/8fb2d0be-87ca-4c04-8020-2087a242612e)

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader - Changes promoted to TEST; tested the same in TEST Environment and all looks good. Here are the results before and after running the batch job.
Before Job Runs - Test Data

![Image](https://github.com/user-attachments/assets/fc819c43-f561-4a2b-9d4f-d312b9504dde)

After Job Runs - Test Results

![Image](https://github.com/user-attachments/assets/f4d5b51d-2b6e-4a49-825d-660eff2cc7b3)

**Comment by TracyLinder** [comment]
@Sreelatha-BlazeKannan please promote to PROD.

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader - PROD DP of #16894 submitted. Service task id:  SCTASK0500044 created. Below are the commit details for PROD promotion.

**Commit Details -** 

Test Commit – 

a)	033b914877b04b76bf45e79d1ceb2350077452d0 – Commit III
b)	5e939e7946c430ae5332be6d9b976e0f0d3c6204 – Commit II

Dev Commit – 

a)	bfcdf4be0322cbdb1c2ea2b33fb5a9ee4e35f549 – Commit III
b)	0bed98bbcec6d1b074e53a01b72a3c261066e630 – Commit II

ATD Local Commit – 

a)	8a1050ca4deee0c8c14c3b69890113850f6f3dd5 – Commit III
b)	e3686b2ed2fa8ba589eb17d6c3dea89922f49d72 – Commit II

Create a batch job to correct DS folders that have not been issued, but have an expiration date.
REPORT_EXPIREDSFOLDER.sql

**Pull Request Link –**

https://github.austintexas.gov/AMANDADevOps/Amandav7_TestingPOC/pull/1554

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader - Service Now task id created **SCTASK0500103** for manual deployment for batch job in PROD.

**Comment by TracyLinder** [comment]
**DTS-AMANDA Convo:**

- The creation of batch job is pending prod promotion. She can speak to Anu at CTM for help. 
- Sree asked for help with test cases. I said i would help since i have PROD permissions. 
- The batch job is programmed to run on Mondays at 12:15am. 


**Test Case Info:**
1. DS, null issue date, expiry date in future

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader - TEST DP Submitted. Created Service Now Task Id: **SCTASK0502029**

**Commit Details –** 

TEST Commit: 47d56236db22e7eefb2f119fa23d1cdadeace624
DEV Commit: 88baa3175d05cdf322526320bfe32a2f0fa924d9
Create a batch job to correct DS folders that have not been issued, but have an expiration date.

DEFAULTREPORT_EXPIREDS
REPORT_EXPIREDSFOLDER.sql
REPORTPROCEDURE

Pull Request Link: https://github.austintexas.gov/AMANDADevOps/Amandav7_TestingPOC/pull/1598

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader - PROD DP of #16894 – 4 for your submitted. Service Now task id: SCTASK0502844 created for the same. 


Commit Details –
PROD Staging Commit: 3b462b1513df11829a2fb7efd0bd0a0ba396cba2
TEST Commit: 47d56236db22e7eefb2f119fa23d1cdadeace624
DEV Commit: 88baa3175d05cdf322526320bfe32a2f0fa924d9
ATD Local Commit: 88baa3175d05cdf322526320bfe32a2f0fa924d9: 
Create a batch job to correct DS folders that have not been issued, but have an expiration date.

DEFAULTREPORT_EXPIREDS
REPORT_EXPIREDSFOLDER.sql
REPORTPROCEDURE

Pull Request Link: 
https://github.austintexas.gov/AMANDADevOps/Amandav7_TestingPOC/pull/1608

**Comment by Nadin-Nader** [comment]
@KyleErich @BondRuss following up to see if the test data was created for PROD?

**Comment by BondRuss** [comment]
@Nadin-Nader can you give us some more details on what you need us to create in PROD for test data? Thank you!
CC: @KyleErich

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader , @KyleErich - 
case1: Few DS Folders Where issue date is Null but not Expiry date and with any status
case2: Few/one DS Folders Where issue date is Null but not Expiry date and with 'expired' status

**Comment by TracyLinder** [comment]
Per Paloma, there are current DS folders in PROD that need to be fixed, so we don't need to create test data in PROD.

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader - DEV DP-5 submitted. Service now ticket id: **SCTASK0505385** created.

**Commit Details -** 

b6899e88349776c5dc238c4c8d8ff34910e37bb7: Create a batch job to correct DS folders that have not been issued, but have an expiration date.

REPORT_EXPIREDSFOLDER.sql

Pull Request Link: 
https://github.austintexas.gov/AMANDADevOps/Amandav7_TestingPOC/pull/1648

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader - As mentioned above, the fix is going to PROD now. Once it goes to PROD, we can run manual test runs with our test data. Upon which we can turn the job ON.

PROD DP of #16894 – 5 submitted. Service now task id: SCTASK0506254  created.

Commit Details - 
PROD Staging Commit: e55eaf7085c0fb9e40631599744b8fe32fefc385
TEST Commit: e55eaf7085c0fb9e40631599744b8fe32fefc385
DEV Commit 
d61fc395994a7b894addfe1ddea02f7bc223f68e
Create a batch job to correct DS folders that have not been issued, but have an expiration date.

REPORT_EXPIREDSFOLDER.sql

https://github.austintexas.gov/AMANDADevOps/Amandav7_TestingPOC/pull/1662/files

**Comment by Nadin-Nader** [comment]
@KyleErich @BondRuss we are going to need a meeting for this as we discovered that there are over 2000 record to be updated.

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader , @TracyLinder  - here is the query 

SELECT f.folderrsn, f.foldertype, vs.statusdesc, f.indate, f.issuedate, f.expirydate FROM folder f 
LEFT JOIN validstatus vs ON vs.statuscode = f.statuscode 
WHERE f.foldertype = 'DS'
AND f.issuedate is NULL
AND f.expirydate IS NOT NULL;

**Comment by BondRuss** [comment]
@TracyLinder @Nadin-Nader @Sreelatha-BlazeKannan 
CC: @KyleErich 
I just received the following report one of the ROW Permitting Supervisors: "**with the ds expiration situation - not sure if this is related, but if you reactivate a permit but do not update the date, it flips back to expired each night. we don't have a way to extend DS permits, we have to reactivate and then it shows as active, and we have to manually extend the date. if it is shown as active, but the date isn't manually updated, then it goes back to expired at night.**"

**Comment by Nadin-Nader** [comment]
@Sreelatha-BlazeKannan  Scheduled a meeting with Jenice and Yuan for 3pm today April 29, 2025

**Comment by Sreelatha-BlazeKannan** [comment]
Kim's email response for Jenice's Concern -
 
The reason we are asking for the expiration date to be removed from DS permits that haven’t been issued is because the permits haven’t been issued. We don’t want the permits to appear active with the populated expiration dates in future. I often have conversations with customers explaining how that expiration date is invalid, as well as the status when the permit changes to an expired status due to this expiration date.

Our rationale for this is that the permit should only show an expiration date when the permit has been issued.

To speak to where or how this field is populated is due to the parent/child relationships that these DS permits have with the BP. It picks up the expiration date from the BP so that they will all expire at the same time. Also, that expiration date for the BP and all children, including DS permits will extend out 180 days from an inspection. This is to keep all the permits active for the customer and encourages them to perform work and call in inspections.

So, we still want this connection to happen, just not until the permit has been issued/activated.

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader - Had meeting with Jenice and Kim on 05/07/2025. Summary is that Jenice agreed to run only for one time to clean up the PROD Data. However need to fix it in code for DS Folder to restrict on Expiry date update. Need to create another card for the fix to resolve programmatically in DS folder itself.

Created a service now ticket # SCTASK0513129 for One time PROD run for tonight.

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader - Had a meeting with Jenice again, we ran the batch job and it completed successfully. Updated the 2153 records as expected. here is the batch job screen shot. 


![Image](https://github.com/user-attachments/assets/31c93541-0d65-43ac-846b-15c31c548e28)

**Comment by TracyLinder** [comment]
### Update

- The initial batch job ran last week.
- On Monday there were 24 expired folders with a future exp. date. Today we see 31. 
    - That means the ongoing issue(s) will create around 10 bad folders a week.
-

**Comment by BondRuss** [comment]
@TracyLinder as of 05/19, the Quality Control Report shows 23 expired folders with a future exp. date. 

[Report.xlsx](https://github.com/user-attachments/files/20395894/Report.xlsx)

**Comment by BondRuss** [comment]
@TracyLinder as of 06/01, the Quality Control Report shows 52 expired DS folders with a future exp. date. 
cc: @KyleErich

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader - Right now it is on hold. As per Tracy, she will be meeting Jenice on 06/11/2025.

**Comment by BondRuss** [comment]
@TracyLinder, as of 06/09, the Quality Control Report shows 38 expired DS folders with a future exp. date. 
cc: @KyleErich 

[Report.xlsx](https://github.com/user-attachments/files/20713115/Report.xlsx)

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader - Waiting for Tracy's update when she meets Jenice on 06/11/2025.

**Comment by TracyLinder** [comment]
Jenice did not join today's meeting; I rescheduled it for Monday 6/16/25

**Comment by TracyLinder** [comment]
@BondRuss @KyleErich we'd like to add a row number limit to this batch job; I'll need ROW to provide that number (we can discuss on Friday). 

Basically the batch job would count the number of DS folders that need to have the expiration date removed, and if the number exceeds our limit the system would abort the batch and send a notification. Based on the recent quality control reports, I think a decent limit would be 50 folders. 

I'll put this on Friday's agenda.

**Comment by TracyLinder** [comment]
### ROW Approves this going to PROD. We also need to update the batch job that emails when a batch job fails, to include this batch job.

@Sreelatha-BlazeKannan please move to PROD, and then work on updating the automated email.

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader - Reverted PROD procedure changes. Created Service now task id:  As the manual job run has to happen first but not the devops deployment. 

Step1 - Revert Process - SCTASK0537536
Step2 - Batch Job update (error email address), Active Batch job n run one time. - SCTASK0537637
Step3 - Re-do/promote devops changes from test to prod

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader - 



SCTASK0538060

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader - Job ran successfully and updated all above records. Right now I see one record like below which created today.

<img width="1162" height="55" alt="Image" src="https://github.com/user-attachments/assets/f45d6a76-21e1-4eae-9776-9700b6d3c889" />

<img width="627" height="72" alt="Image" src="https://github.com/user-attachments/assets/03a7402b-0520-482a-a4a5-d4f51d9b2dd0" />

**Comment by TracyLinder** [comment]
As an FYI the DS folder in Sree's screenshot above was created with an expiration date when the PR folder was approved today.

**Comment by BondRuss** [comment]
@TracyLinder do we think that the PR folder(s) process attempt and status changes are going to override this batch job?

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader - As of Today there are 15 records to be fixed in this week's batch job.

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/SREELA~1/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/SREELA~1/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#467886" vlink="#96607D">


Sno | Folder RSN | In Date | Issue Date | Expiry Date | Status
-- | -- | -- | -- | -- | --
1 | 13561814 | 8/5/2025 |   | 2/3/2026 | Pending
2 | 13561110 | 8/4/2025 |   | 2/3/2026 | Pending
3 | 13561827 | 8/5/2025 |   | 2/3/2026 | Pending
4 | 13560458 | 8/4/2025 |   | 7/1/2026 | Pending
5 | 13559632 | 8/1/2025 |   | 2/1/2026 | Pending
6 | 13490065 | 3/25/2025 |   | 2/1/2026 | Pending
7 | 13561074 | 8/4/2025 |   | 2/3/2026 | Pending
8 | 13561389 | 8/5/2025 |   | 2/2/2026 | Pending
9 | 13561175 | 8/4/2025 |   | 2/2/2026 | Pending
10 | 13562833 | 8/6/2025 |   | 3/30/2026 | Pending
11 | 13560877 | 8/4/2025 |   | 6/29/2026 | Pending
12 | 13561413 | 8/5/2025 |   | 2/2/2026 | Pending
13 | 13560817 | 8/4/2025 |   | 5/11/2026 | Pending
14 | 13561401 | 8/5/2025 |   | 2/2/2026 | Pending
15 | 13562959 | 8/6/2025 |   | 2/2/2026 | Pending



</body>

</html>

**Comment by TracyLinder** [comment]
This job is stable; I am closing.
